app/models/concerns/orchestration/dns.rb

```diff
@@ -7,11 +7,15 @@ module Orchestration::DNS
   end
 
   def dns?
-    hostname.present? && ip_available? && !domain.nil? && !domain.proxy.nil? && host.managed? && managed?
+    # host.managed? and managed? should always come first so that orchestration doesn't
+    # even get tested for such objects
+    (host.nil? || host.managed?) && managed? && hostname.present? && ip_available? && !domain.nil? && !domain.proxy.nil?
```

Comment:
that's good catch, if mac or ip would be nil, it would fail, we need the `host.nil?` check there as well